vmm/src/lib.rs

```diff
@@ -2267,15 +2267,37 @@ mod tests {
     #[test]
     fn test_epoll_stdin_event() {
         let mut epoll_context = EpollContext::new().unwrap();
-        epoll_context.enable_stdin_event();
-        assert_eq!(
-            epoll_context.dispatch_table[epoll_context.stdin_index as usize].unwrap(),
-            EpollDispatch::Stdin
-        );
-        // This should trigger a warn!. When logger will not be a global var we will be able to test
-        // it.
-        epoll_context.enable_stdin_event();
 
+        // If this unit test is run without a terminal attached (i.e ssh without pseudo terminal,
+        // request, jailer with `--deamonize` flag on) EPOLL_CTL_ADD would return EPERM
+        // on STDIN_FILENO.
+        // Here we are testing that if we receive EPERM, we are no longer asserting
+        // against `epoll_context.dispatch_table[epoll_context.stdin_index as usize]` holding any value.
```

Comment:
Considering these restrictions, is this test ever going to be run by the CI?

Reply:
Yes, the test is run by the CI both on x86_64 and aarch64.
The x86_64 hits the first branch of the match (i.e `Ok`) while aarch64 hits the EPERM branch.
The other solution to this problem would be to run the tests with `ssh -t` (which allocates a pseudo terminal by force) but i did not choose that path because it contravenes with the `--daemonize` flag where we want firecracker to redirect all three standard I/O file descriptors to dev/null.

Reply:
This is not a new one, it is also described at `https://github.com/firecracker-microvm/firecracker/blob/master/vmm/src/lib.rs#L211`.

Reply:
I see. My worry was that this test can silently fail and hide away problems with our code.